packages/block-library/src/button/style.scss

```diff
@@ -36,6 +36,10 @@ $blocks-button__height: 56px;
 	}
 }
 
+.wp-gs .wp-block-button__link:not(.has-background) {
```

Comment:
Can you explain to me why the `.wp-gs` is necessary? Why can't we just drop it?

Reply:
If I added the rule `.wp-block-button__link:not(.has-background)` to the block most existing themes would break. That happens because that selector has a specificity of two classes, so a theme setting the background color using `.wp-block-button__link{ background-color: ...}` would stop working.

Another case where this problem is even more noticeable is the implementation of global styles in a paragraph. We can not simply set `p { background-color:  var(--wp-gs-paragraph-color-text, var( --wp-gs-color-text ) )` otherwise paragrapgs would probably change colors in most existing themes. 

The `wp-gs` class offers a block the possibility of keeping existing styles so existing themes styling them still work while at the same time they could provide styles that are used in the presence of a theme compatible with global styles.

The CSS variables with the defaults or user-configured options will be always be loaded even if the theme does not support global styles. So a new block build now may not need to use wp-gs and can just use the global styles all the time.

Reply:
Can these problems be solved if we avoid an explicit value at the last argument. Concretely, if it's something like that

```
background-color: var(--wp-core-button-color-background, var(--wp-color-primary, initial));
```

(initial or inherit depending on the property we're setting)

Reply:
No, this code would not work because --wp-color-primary would always be defined as it is a default that WordPress needs to provide even if themes don't use global styles. If defaults were not provided blocks could never safely rely on a global variable.

Reply:
>  --wp-color-primary would always be defined as it is a default that WordPress needs 

Making sure I'm understanding properly here. Why WordPress need to define defaults for these variables?

Reply:
in other words, why can't they just be "undefined" if a theme doesn't have any config and we don't have any stored config?

Reply:
I think it would be good to provide a set of defaults to make things consistent even if the theme is not setting global styles.
So, for example, block A and block B created by different authors both use the --wp-gs-color-text, if a default for the variable is not provided both blocks may end up using different text colors while ideally, the text color would be the same.

Reply:
Why not just consider the browser defaults as the defaults? There's something I'm missing here.

If these blocks do support the variables they will have something like that on their styles right?

```
color: var(--wp-block-something-color-primary, var(--wp-color-primary, initial));
```

Reply:
If we set something like `p:not(.has-background) { color: var(--wp-block-something-color-primary, var(--wp-color-primary, initial)); }` without targetting current themes supporting global styles (having wp-gs class) we make break the styles for example if the current theme has `p { color: red }` the style will stop working as color will take the default browser value.

Reply:
What happens if we just remove "initial" from that code?

Reply:
> What happens if we just remove "initial" from that code?

In my tests, I think the browser uses the default value for the property when the var is used (it seems the same as including initial).

So I think we can not avoid wp-gs otherwise we may break styles.